File: rust/src/169_majority_element.rs
impl Solution {
    pub fn majority_element(nums: Vec<i32>) -> i32 {
        let mut maj = 0;
        let mut count = 0;
        for num in &nums {
            if count == 0 {
                maj = *num;
            }
            if maj != *num {
                count -= 1;
            } else {
                count += 1;
            }
        }
        return maj;
    }
}

File: c/169_majority_element.c
int majorityElement(int* nums, int numsSize) {
    int majority;
    int count = 0;
    for (int i = 0; i < numsSize; i++) {
        if (count == 0) {
            majority = nums[i];
        }
        if (majority == nums[i]) {
            count++;
        } else {
            count--;
        }
    }
    return majority;
}
